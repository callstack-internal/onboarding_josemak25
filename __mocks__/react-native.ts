jest.mock("react-native", () => {
  const RN = jest.requireActual("react-native"); // use original implementation, which comes with mocks out of the box

  // mock modules/components created by assigning to NativeModules
  RN.NativeModules.SourceCode = {
    scriptURL: "http://localhost:3000/",
  };

  return RN;
});
