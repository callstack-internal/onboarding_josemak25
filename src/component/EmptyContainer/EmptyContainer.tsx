import React, { FC } from "react";
import { StyleProp, View, Text, ViewStyle } from "react-native";

import Ionicons from "react-native-vector-icons/Ionicons";

Ionicons.loadFont(); // initialize font icons

import { makeUseStyles } from "../../helpers/makeUseStyles";

type EmptyContainerProps = {
  title: string;
  size?: number;
  subtitle?: string;
  style?: StyleProp<ViewStyle>;
};

export const EmptyContainer: FC<EmptyContainerProps> = ({
  title,
  style,
  subtitle,
  size = 38,
}) => {
  const { styles, palette } = useStyles();

  return (
    <View style={[styles.container, style]}>
      <View style={styles.iconContainer}>
        <Ionicons name="refresh" size={size} color={palette.text} />
      </View>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.subtitle}>{subtitle}</Text>
    </View>
  );
};

const useStyles = makeUseStyles(({ palette, layout, fonts }) => ({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    marginHorizontal: layout.gutter * 2,
  },
  iconContainer: {
    width: 86,
    height: 86,
    borderWidth: 2,
    borderRadius: 43,
    alignItems: "center",
    justifyContent: "center",
    borderColor: palette.text,
  },
  title: {
    textAlign: "center",
    color: palette.text,
    fontSize: fonts.size.lg,
    marginVertical: layout.gutter,
  },
  subtitle: {
    color: palette.text,
    textAlign: "center",
  },
}));
