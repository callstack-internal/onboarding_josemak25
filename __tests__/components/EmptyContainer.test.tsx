import React from "react";

import { render, screen } from "@testing-library/react-native";
import { EmptyContainer } from "../../src/component/EmptyContainer";

describe("components / EmptyContainer.tsx", () => {
  const options = { title: "No Weather Data", subtitle: "Pull-to-refresh" };

  it("should render correctly", () => {
    render(<EmptyContainer {...options} />);

    expect(screen.getByText(options.title)).toBeTruthy();
    expect(screen.getByText(options.subtitle)).toBeTruthy();
  });
});
