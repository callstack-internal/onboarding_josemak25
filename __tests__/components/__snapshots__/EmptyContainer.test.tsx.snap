// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components / EmptyContainer.tsx should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginHorizontal": 24,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#000",
        "borderRadius": 43,
        "borderWidth": 2,
        "height": 86,
        "justifyContent": "center",
        "width": 86,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 38,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï’½
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 20,
        "marginVertical": 12,
        "textAlign": "center",
      }
    }
  >
    No Weather Data
  </Text>
  <Text
    style={
      Object {
        "color": "#000",
        "textAlign": "center",
      }
    }
  >
    Pull-to-refresh
  </Text>
</View>
`;
